_posts/2017/2017-04-06-javaland-2017-in-phantasialand.md

```diff
@@ -0,0 +1,75 @@
+Better change your could instead of using these flags.
+If you want to migrate to use modules you can put a plain jair on the module path (automatic modules).
+The name is derived from the jar name and export everthing needed as well as reads other downstream modules.
+It can access the class path. You should create a module which requires `commons.lang` and a `module-info.jar`.
+Dependencies have to be module themselves for easier handling (transitive dependencies).
+At runtime services (components) are accessible via reflection.
+Don’t open module to everyone.
+Export only certain api packages including services!
+Revisit open modules when you are done.
+Create several modules that are logically seperated, e.g. `API`, `Impl` and `Main`.
+
+# Workshop Day
+
+At the [workshop day](https://www.javaland.eu/de/programm/schulungstag) we visited the Java Web Security session by [Dominik Shadow](https://twitter.com/dschadow) ([Website](https://blog.dominikschadow.de/)) – material can be found [here](https://github.com/dschadow/JavaSecurity) – and the Hibernate 4 session by [Thorben Jansen](https://twitter.com/thjanssen123), who just released a new [Hibernate Tips book](http://www.thoughts-on-java.org/hibernate-tips-book-release-special-launch-price/) about it and run a very popular web portal about Java and Hibernate - [www.thoughts-on-java.org](http://www.thoughts-on-java.org/).
+Both sessions were an absolute blast and highlights of the conference itself as both trainers had many years of coding and lecturing experience.
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">AWESOME workshop + FULL room  <a href="https://twitter.com/JavaLandConf">@JavaLandConf</a> <a href="https://twitter.com/hashtag/Javaland?src=hash">#Javaland</a>: <a href="https://twitter.com/thjanssen123">@thjanssen123</a> teaches <a href="https://twitter.com/Hibernate">@Hibernate</a> 4 <a href="https://twitter.com/hashtag/JPA?src=hash">#JPA</a> <a href="https://twitter.com/hashtag/ORM?src=hash">#ORM</a> <a href="https://twitter.com/hashtag/lucene?src=hash">#lucene</a> <a href="https://twitter.com/hashtag/bridges?src=hash">#bridges</a> <a href="https://t.co/RrG0LWycGb">pic.twitter.com/RrG0LWycGb</a></p>&mdash; Benjamin Nothdurft (@dataduke) <a href="https://twitter.com/dataduke/status/847394050986594307">March 30, 2017</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+# People, Fun and Food
+
+The conference is all about meeting great Java software engineers from other countries as well, including real [Java Champions](https://community.oracle.com/community/java/java-champions) and [JavaOne Rockstars](https://www.oracle.com/javaone/rock-star-wall-of-fame.html), and meeting Java User Groups from Germany (see [iJUG](ijug.eu), the umbrella organisation of many [JUG_DE](https://twitter.com/JUG_DE)) from all around the world.
```

Comment:
Apart from the amazing talks and workshops, the conference is also about meeting....